h2o-r/h2o-package/R/models.R

```diff
@@ -2969,6 +3043,37 @@ extract_scoring_history <- function(model, value) {
   }
 }
 
+#'
+#' Return the GLM coefficient names without building the actual GLM model by setting max_iterations=0.
+#'
+#' @param object an \linkS4class{H2OModel} object.
+#' 
+#' @examples 
+#' \dontrun{
+#' library(h2o)
+#' h2o.init()
+#' 
+#' f <- "https://s3.amazonaws.com/h2o-public-test-data/smalldata/junit/cars_20mpg.csv"
+#' cars <- h2o.importFile(f)
+#' predictors <- c("displacement", "power", "weight", "acceleration", "year")
+#' response <- "cylinders"
+#' cars_glm <- h2o.glm(balance_classes = TRUE, 
+#'                     seed = 1234, 
+#'                     x = predictors, 
+#'                     y = response, 
+#'                     training_frame = cars,
+#'                     max_iterations=0)
+#' h2o.coef_names(cars_glm)
+#' }
+#' @export
+h2o.coef_names <- function(object) {
+    if (is(object, "H2OModel") &&
+        (object@algorithm %in% c("glm"))) {
+        coef_names = object@model$coefficient_names
+        return(coef_names[-length(coef_names)])
+    }
```

Comment:
In python you remove the `Intercept` here you remove the last element. This is wrong I think:
1) AFAIK you can have GLM without the `Intercept`
2) The intercept might not always be the last entry in the beta array. (I think this is true for HGLM (https://github.com/h2oai/h2o-3/blob/d1b4f3c79d9f4921579ce69ef41a9b58a94f20cd/h2o-algos/src/main/java/hex/glm/ComputationState.java#L240))

Reply:
HGLM is not working properly and I don't want to care about it at this point.  Convention in the code is the intercept is at the end.  I did add a check to make sure if intercept is disabled, the last element will not be dropped.